server/client.go

```diff
@@ -3705,6 +3705,13 @@ func (c *client) processMsgResults(acc *Account, r *SublistResult, msg, deliver,
 			}
 			continue
 		}
+
+		// If this sub is an internal account subscription, meaning a service import, and this
+		// message has already been mapped via a previous service import, skip.
+		if flags&pmrMsgImportedFromService != 0 && sub.icb != nil && sub.client.kind == ACCOUNT {
+			continue
```

Comment:
Not an actual code change but maybe add a todo in all the places in this and the previous PR so that in a subsequent change we can set a flag in either the client or the account to maybe record this event?
Similar to incomplete in the account. If a cycle is the only way this can happen then maybe the account.
if this can happen for some other reason (say reconnect) maybe in the client.

Reply:
Now that I am looking at my next problem I think this won't work since we do want to daisy chain service imports, we just do not want cycles.

Reply:
So I may go back to what I had before and try to expand to detect complex loops.

Reply:
> Now that I am looking at my next problem I think this won't work since we do want to daisy chain service imports, we just do not want cycles.

@derekcollison wouldn't you want to go hub/spoke vs daisy chain? Seems to me that the workaround would be to import from the original account directly. What's the daisy chain use case, proxy accounts, easy forwarding from private to public imports?

Reply:
Daisy chain is export $JS.API.>. Will go from account A to account B to $SYS.